pkg/startup/startup.go

```diff
@@ -720,6 +742,15 @@ func configureIPPools(ctx context.Context, client client.Interface) {
 		return
 	}
 
+	// If CIDRs weren't specified through the environment variables, check if they're present in kubeadm's
+	// config map.
+	if (len(ipv4Pool) == 0 || len(ipv6Pool) == 0) && kubeadmConfig != nil {
+		ipv4Pool, ipv6Pool, err := extractKubeadmCIDRs(kubeadmConfig)
+		if err == nil {
```

Comment:
Here's another case where we're not handling an error. We need an `err != nil` case here.